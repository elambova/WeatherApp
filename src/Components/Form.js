import React, { Component } from "react";
import PropTypes from "prop-types";

export default class Form extends Component {
  state = {
    city: "",
  };

  static propTypes = {
    handleSubmit: PropTypes.func.isRequired,
  };

  onSubmit = async (e) => {
    e.preventDefault();
    if (this.props.handleSubmit) {
      this.props.handleSubmit(this.state.city);
    }
  };
  render() {
    return (
      <React.Fragment>
        <form method="POST" onSubmit={this.onSubmit}>
          <div>
            <label htmlFor="city">Please enter city:</label>
            <input
              type="text"
              placeholder="City"
              name="city"
              id="city"
              onChange={(e) => this.setState({ city: e.target.value })}
              required
            />
          </div>
          <input type="submit" id="submit-btn" value="Enter" name="submit" />
        </form>
      </React.Fragment>
    );
  }
}
