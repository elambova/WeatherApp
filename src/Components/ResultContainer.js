import React, { Component } from "react";
import PropTypes from "prop-types";

export default class ResultContainer extends Component {
  static propTypes = {
    data: PropTypes.object.isRequired,
  };

  render() {
    const { data } = this.props;
    const date = new Date().toDateString();

    return (
      <React.Fragment>
        <h2>{data.city} </h2>
        <h3 className="date">{date}</h3>

        <img
          src={
            data.weather.icon !== undefined
              ? `https://www.weatherbit.io/static/img/icons/${data.weather.icon}.png`
              : `http://openweathermap.org/img/wn/${data.weather[0].icon}@2x.png`
          }
          alt={`${data.weather.description}`}
        />
        <p>{data.weather.description}</p>
        <p>
          Temperature: <strong>{data.temp}</strong> &#8451;
        </p>
        <img
          id="city-pic"
          src={`${data.picture}`}
          alt={`${data.city}, ${data.country}`}
        />
        <p className="tags-image">{data.tags}</p>
      </React.Fragment>
    );
  }
}
