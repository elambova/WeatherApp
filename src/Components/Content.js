import React, { Component } from "react";
import ResultContainer from "./ResultContainer";
import Form from "./Form";
import PropTypes from "prop-types";
import Loader from "react-loader-spinner";

export default class Content extends Component {
  static propTypes = {
    handleSubmit: PropTypes.func.isRequired,
    data: PropTypes.object.isRequired,
  };

  render() {
    const { data, handleSubmit } = this.props;
    return (
      <div id="content">
        <Form handleSubmit={handleSubmit} />
        <div id="result">
          {Object.keys(data).length !== 0 ? (
            <ResultContainer data={data} />
          ) : (
            <Loader type="Oval" color="teal" height={80} width={80} />
          )}
        </div>
      </div>
    );
  }
}
