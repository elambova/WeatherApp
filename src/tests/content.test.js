import React from "react";
import Enzyme, { mount, shallow } from "enzyme";
import Content from "../Components/Content";
import Form from "../Components/Form";
import Adapter from "enzyme-adapter-react-16";
import ResultContainer from "../Components/ResultContainer";
import toJson from "enzyme-to-json";

Enzyme.configure({ adapter: new Adapter() });
let content;

const handleSubmit = () => {
  return "SUBMIT";
};

const data = {
  city: "Sofia",
  country: "Bulgaria",
  picture:
    "https://pixabay.com/get/54e2d3474c54ad14f1dc84609629317f123cd9e4574c704c7c297ad49449c250_640.jpg",
  weather: {
    icon: "c01d",
    description: "Clear sky",
  },
};

beforeEach(() => {
  content = mount(<Content data={data} handleSubmit={handleSubmit} />);
});

describe("App container", () => {
  test("should be exists", () => {
    expect(content.exists()).toBe(true);
  });

  test("render Form Container", () => {
    const formContainer = mount(<Form handleSubmit={handleSubmit} />);
    expect(formContainer.props()).toMatchSnapshot();
  });

  test("render Result Container", () => {
    const resultContainer = mount(<ResultContainer data={data} />);

    expect(resultContainer.props("data")).toMatchSnapshot();
  });

  test("Form Container simulate submit", () => {
    const formContainer = mount(<Form handleSubmit={handleSubmit} />);
    formContainer.simulate("submit");
    expect(formContainer.prop("handleSubmit")()).toBe("SUBMIT");
  });

  test("check in Result Container data.city is Sofia", () => {
    const resultContainer = mount(<ResultContainer data={data} />);

    expect(resultContainer.props("data").data.city).toBe("Sofia");
  });

  test("renders without crashing given the required props", () => {
    const form = shallow(<Form handleSubmit={handleSubmit} />);

    expect(toJson(form)).toMatchSnapshot();
  });

  test("required props type Function", () => {
    const form = shallow(<Form handleSubmit={handleSubmit} />);

    expect(toJson(form).node.nodeType).toBe("function");
  });
});
