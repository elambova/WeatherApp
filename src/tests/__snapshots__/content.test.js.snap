// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App container render Form Container 1`] = `
Object {
  "handleSubmit": [Function],
}
`;

exports[`App container render Result Container 1`] = `
Object {
  "data": Object {
    "city": "Sofia",
    "country": "Bulgaria",
    "picture": "https://pixabay.com/get/54e2d3474c54ad14f1dc84609629317f123cd9e4574c704c7c297ad49449c250_640.jpg",
    "weather": Object {
      "description": "Clear sky",
      "icon": "c01d",
    },
  },
}
`;

exports[`App container renders without crashing given the required props 1`] = `
<Fragment>
  <form
    method="POST"
    onSubmit={[Function]}
  >
    <div>
      <label
        htmlFor="city"
      >
        Please enter city:
      </label>
      <input
        id="city"
        name="city"
        onChange={[Function]}
        placeholder="City"
        required={true}
        type="text"
      />
    </div>
    <input
      id="submit-btn"
      name="submit"
      type="submit"
      value="Enter"
    />
  </form>
</Fragment>
`;
